packages/core/src/decorators/transport/TransportDecorator.test.ts

```diff
@@ -4,7 +4,7 @@ import { MessageValidator } from '../../utils/MessageValidator'
 import { TransportDecorator, ReturnRouteTypes } from './TransportDecorator'
 
 const validTransport = (transportJson: Record<string, unknown>) =>
-  MessageValidator.validate(JsonTransformer.fromJSON(transportJson, TransportDecorator))
+  MessageValidator.validate(JsonTransformer.fromJSON(transportJson, TransportDecorator, { validate: false }))
```

Comment:
Same here, if we keep validation would be nice

Reply:
similar to above - it doesn't pass validation. Happy correct this somehow. Will need to do some digging though because I'm not familiar enough with AFJ to be able to tell what's going on/wrong (the test says it should transform this - so that makes me assume it should pass validation also). Maybe I didn't get the validation quite right...

Reply:
Can you share the error(s) you get here?

Reply:
Here the output for validation enabled:

```
FAIL   @aries-framework/core  packages/core/src/decorators/transport/TransportDecorator.test.ts
  Decorators | TransportDecorator
    ✓ should correctly transform Json to TransportDecorator class (2 ms)
    ✓ should correctly transform TransportDecorator class to Json
    ✕ should only allow correct return_route values (2 ms)
    ✕ should require return_route_thread when return_route is thread

  ● Decorators | TransportDecorator › should only allow correct return_route values

    ClassValidationError: function: Failed to validate class.

      26 |     const errors = validateSync(classInstance)
      27 |     if (isValidationErrorArray(errors)) {
    > 28 |       throw new ClassValidationError('Failed to validate class.', {
         |             ^
      29 |         classType: typeof Class,
      30 |         validationErrors: errors,
      31 |       })

      at Function.validateSync (src/utils/MessageValidator.ts:28:13)
      at Function.fromJSON (src/utils/JsonTransformer.ts:34:24)
      at validTransport (src/decorators/transport/TransportDecorator.test.ts:7:45)
      at expectInvalid (src/decorators/transport/TransportDecorator.test.ts:11:10)
      at Object.<anonymous> (src/decorators/transport/TransportDecorator.test.ts:68:11)

  ● Decorators | TransportDecorator › should only allow correct return_route values

    expect.assertions(4)

    Expected four assertions to be called but received three assertion calls.

      62 |
      63 |   it('should only allow correct return_route values', async () => {
    > 64 |     expect.assertions(4)
         |            ^
      65 |     await expectValid(valid.all)
      66 |     await expectValid(valid.none)
      67 |     await expectValid(valid.thread)

      at Object.<anonymous> (src/decorators/transport/TransportDecorator.test.ts:64:12)

  ● Decorators | TransportDecorator › should require return_route_thread when return_route is thread

    ClassValidationError: function: Failed to validate class.

      26 |     const errors = validateSync(classInstance)
      27 |     if (isValidationErrorArray(errors)) {
    > 28 |       throw new ClassValidationError('Failed to validate class.', {
         |             ^
      29 |         classType: typeof Class,
      30 |         validationErrors: errors,
      31 |       })

      at Function.validateSync (src/utils/MessageValidator.ts:28:13)
      at Function.fromJSON (src/utils/JsonTransformer.ts:34:24)
      at validTransport (src/decorators/transport/TransportDecorator.test.ts:7:45)
      at expectInvalid (src/decorators/transport/TransportDecorator.test.ts:11:10)
      at Object.<anonymous> (src/decorators/transport/TransportDecorator.test.ts:74:11)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 2 passed, 4 total
Snapshots:   0 total
Time:        0.294 s, estimated 1 s
Ran all test suites matching /packages\/core\/src\/decorators\/transport\/TransportDecorator.test.ts/i.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Can you share the actual validation errors? Also can we look at extending the error message to include the actual validation errors? It would be really useful to see the valiadtion error in the message/stack.

Reply:
yes will do

Reply:
alright i figured it out. Will push in a second.

Reply:
Regarding the `It would be really useful to see the validation error in the message/stack.` if I understand that correctly, I added that already last night. You can now either use the validationErrors attribute on ClassValidationError object or you can call the inspect() method which also provides the errors

Reply:
I mean it would be useful to somehow print the validation error when the error/stack is printed. 

See this thread for an example of what I mean: https://github.com/typestack/class-validator/issues/26#issuecomment-374059419

It appears validation error exposes an `toString()` method that you can call. If we loop over the array and print it for each error that would be nice I think.